### PR TITLE
JSON.parse and JSON.stringify

### PR DESCRIPTION
**Pull request by NickNaso**
Hi everyone,
I tried to add `Napi::JSON::Parse` and `Napi::JSON::Stringify` functions like discussed in the following issue **[https://github.com/nodejs/node-addon-api/issues/66](https://github.com/nodejs/node-addon-api/issues/66)**
This PR is only a start I want feedback about the created interface and implementation.

### CODE REVIEW

File: napi-inl.h [hunk 1]
```diff
@@ -3453,6 +3453,30 @@ inline const napi_node_version* VersionManagement::GetNodeVersion(Env env) {
...
+////////////////////////////////////////////////////////////////////////////////
+// JSON class
+////////////////////////////////////////////////////////////////////////////////
+inline Value JSON::Parse(Env env, String json_string) {
+  Object json = env.Global().Get("JSON").As<Object>();
+  Function parse = json.Get("parse").As<Function>();
+  return parse.Call(json, { json_string });
+}
+
+inline Value JSON::Stringify(Env env,
```
Comment (romandev) [inline napi-inl.h]: I'm not sure but return `String` type might be better than `Value`. WDYT?
(FYI, ToString() returns `String` type)

File: napi-inl.h [hunk 2]
```diff
@@ -3453,6 +3453,30 @@ inline const napi_node_version* VersionManagement::GetNodeVersion(Env env) {
   return result;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// JSON class
+////////////////////////////////////////////////////////////////////////////////
+inline Value JSON::Parse(Env env, String json_string) {
```
Comment (romandev) [inline napi-inl.h]: I'm not sure but return `Object` type might be better than `Value` type. WDYT?

File: test/binding.gyp [hunk 3]
```diff
@@ -17,6 +17,7 @@
         'error.cc',
         'external.cc',
         'function.cc',
+        'json.cc',
```
Comment (romandev) [inline test/binding.gyp]: nit: alphabetical order?

File: test/index.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ let testModules = [
   'error',
   'external',
   'function',
+  'json',
```
Comment (romandev) [inline test/index.js]: nit: alphabetical order?

File: napi-inl.h (hunk 2 above)
Comment (devsnek) [reply napi-inl.h]: it can return an object, array, string, number, or boolean

Comment (romandev) [reply napi-inl.h]: Ah, you're right! I was wrong. Thank you for correcting me.
@NickNaso, please ignore my previous comment.

Comment (NickNaso) [reply napi-inl.h]: Ok thanks! 👍

File: test/binding.gyp (hunk 3 above)
Comment (NickNaso) [reply test/binding.gyp]: Ok fixed!

File: napi-inl.h (hunk 1 above)
Comment (NickNaso) [reply napi-inl.h]: Yes you're right.

**Comment by NickNaso** [comment]
@romandev @devsnek Hi made the requested changes. Please when you have time take a look.

**Comment by NickNaso** [comment]
@devsnek you're right, with this implementation I want try to find what could be the interface to expose  and then release it. 
After that my intention is to continue work creating a completely native function like V8 and other JS engines do.

**Comment by NickNaso** [comment]
Hi everyone I'm closing this PR. I will continue to work on **JSON.parse** and **JSON.stringify**  to provide these feature in an additional package for **node-addon-api**. See https://github.com/nodejs/abi-stable-node/issues/363

**Pull request closed without merging**